app/controllers/stats_controller.rb

```diff
@@ -122,9 +150,10 @@ def create_spreadsheet(report_data)
   def set_fields_for_custom_action
     @custom_reports_foi = ReportType.custom.foi
     @custom_reports_sar = ReportType.custom.sar
+    @custom_reports_offender_sar = ReportType.custom.offender_sar
     @custom_reports_closed_cases = ReportType.closed_cases_report
-    @correspondence_types = CorrespondenceType.custom_reporting_types +
-      [self.class.closed_cases_correspondence_type]
+    @correspondence_types = CorrespondenceType.custom_reporting_types & current_user.permitted_correspondence_types
```

Comment:
Cool! So this is using the "Set Intersection" operator from ArrayClass? Just checking because there are a few uses of unary ampersand in Ruby and I think most people (myself included) find it hard to keep them straight. 
https://ablogaboutcode.com/2012/01/04/the-ampersand-operator-in-ruby

Reply:
I wonder if this line is worth pulling out into a method to help explain what's going on here

Reply:
@Polsonby  could do,  basically the user can only generate the custom report within his/her permitted scope